### PR TITLE
Fix build when enabling mdebug options.

### PR DESCRIPTION
**Pull request by richsalz**

### CODE REVIEW

**Comment by romen** [comment]
I can confirm this fixed all the issues I detected and reported in #10572!

File: crypto/mem.c [hunk 1]
```diff
@@ -305,6 +297,24 @@ int CRYPTO_mem_debug_pop(void)
     return -1;
 }
 
+void CRYPTO_mem_debug_malloc(void *addr, size_t num, int flag,
```
Comment (romen) [inline crypto/mem.c]: Should these functions be surrounded by `#ifdef` deprecation guards?

Comment (richsalz) [reply crypto/mem.c]: look up a few dozen lines.

Comment (richsalz) [reply crypto/mem.c]: and not that @levitte approved this :)

Comment (richsalz) [reply crypto/mem.c]: Oops, typo.  *Note* that Richard approved this.

Comment (romen) [reply crypto/mem.c]: > look up a few dozen lines.

oh yes, I completely missed it

https://github.com/openssl/openssl/pull/10629/files#diff-0ad35796f3b4040e66893ac4cb4b6b28R274-R276




> _Note_ that Richard approved this.

Sometimes even @levitte can overlook things, that's why I was asking for confirmation ;)

**Comment by slontis** [comment]
Thanks. Merged to master.
A Fixes tag might have been good - will manually close the issue

**Pull request closed without merging**